### PR TITLE
add a new method to model_manager that retrieves individual pipeline components

### PR DESCRIPTION
**Pull request by lstein**
This PR introduces a new set of ModelManager methods that enables you to retrieve the individual parts of a stable diffusion pipeline model, including the vae, text_encoder, unet, tokenizer, etc.

To use:

```
from invokeai.backend import ModelManager

manager = ModelManager('/path/to/models.yaml')

# get the VAE
vae = manager.get_model_vae('stable-diffusion-1.5')

# get the unet
unet = manager.get_model_unet('stable-diffusion-1.5')

# get the tokenizer
tokenizer = manager.get_model_tokenizer('stable-diffusion-1.5')

# etc etc
feature_extractor = manager.get_model_feature_extractor('stable-diffusion-1.5')
scheduler = manager.get_model_scheduler('stable-diffusion-1.5')
text_encoder = manager.get_model_text_encoder('stable-diffusion-1.5')

# if no model provided, then defaults to the one currently in GPU, if any
vae = manager.get_model_vae()
```

### CODE REVIEW

**Comment by blessedcoolant** [comment]
Code reads fine to me. Pretty straight forward. Do I need to test any use cases here?

File: invokeai/backend/model_management/model_manager.py [hunk 1]
```diff
@@ -135,6 +169,33 @@ def get_model(self, model_name: str=None)->dict:
             "hash": hash,
         }
 
+    def get_sub_model(
```
Comment (Kyle0654) [inline invokeai/backend/model_management/model_manager.py:172]: I think I'd almost prefer a separate method for each of these. Would make the usage a lot more clear, and avoid needing to manually set type hints on the user's end.

**Review by Kyle0654 — CHANGES_REQUESTED** [review]

File: invokeai/backend/model_management/model_manager.py (hunk 1 above)
Comment (lstein) [reply invokeai/backend/model_management/model_manager.py:172]: Ok, sure. Will do.

Comment (lstein) [reply invokeai/backend/model_management/model_manager.py]: This is now done.

**Comment by lstein** [comment]
> Code reads fine to me. Pretty straight forward. Do I need to test any use cases here?

I don't think so. I ran through all the test cases and didn't find any problems.

**Review by blessedcoolant — APPROVED** [review]

**Review by Kyle0654 — APPROVED** [review]

**Pull request merged**